test/rules/input_object_values_are_camel_cased.js

```diff
@@ -0,0 +1,54 @@
+import assert from 'assert';
+import { parse } from 'graphql';
+import { validate } from 'graphql/validation';
+import { buildASTSchema } from 'graphql/utilities/buildASTSchema';
+
+import { InputObjectValuesAreCamelCased } from '../../src/rules/input_object_values_are_camel_cased';
+
+describe('InputObjectValuesAreCamelCased rule', () => {
+  it('catches input object type values that are not camel cased', () => {
+    const ast = parse(`
+      type Query {
+        hello: String
+      }
+
+      input User {
+        user_name: String
+
+        withDescription: String
+      }
+    `);
+
+    const schema = buildASTSchema(ast);
+    const errors = validate(schema, ast, [InputObjectValuesAreCamelCased]);
+
+    assert.equal(errors.length, 1);
+
+    assert.equal(errors[0].ruleName, 'input-object-values-are-camel-cased');
+    assert.equal(
+      errors[0].message,
+      'The input value `User.user_name` is not camel cased.'
+    );
+    assert.deepEqual(errors[0].locations, [{ line: 7, column: 9 }]);
+  });
+
+  it('catches arguments that are not camel cased', () => {
+    const ast = parse(`
+      type Query {
+        hello(argument_without_description: String): String
+      }
+    `);
+
+    const schema = buildASTSchema(ast);
+    const errors = validate(schema, ast, [InputObjectValuesAreCamelCased]);
+
+    assert.equal(errors.length, 1);
+
+    assert.equal(errors[0].ruleName, 'input-object-values-are-camel-cased');
+    assert.equal(
+      errors[0].message,
+      'The input value `hello.argument_without_description` is not camel cased.'
```

Comment:
I just looked at the changes again and noticed this test.

I was a bit surprised that the `InputObjectValuesAreCamelCased` reports arguments that are not camel cased. I'm thinking we might want that to be part of a different rule.

It looks like the reason why this is happening is because `InputValueDefinition` node is used by the parser for both `InputFieldsDefinition` and `ArgumentsDefinition` nodes.

i.e.

```js
ArgumentsDefinition : ( InputValueDefinition+ )
InputFieldsDefinition : { InputValueDefinition+ }
```

We should be able to get around this by making the rule ignore input values that are within a `ArgumentsDefinition` by adding this:

```js
ArgumentsDefinition(node) {
  return false;
},
```

I would then change this test to show that the rule doesn't validate arguments.

Reply:
I do not understand why we would want to do this. This PR is to close #28 and #67 .

Another way to filter arguments that you have also used in the codebase already is:
```
if (parentNode.kind != 'InputObjectTypeDefinition') {
        return;
      }
```

Unless you want to have different rules, for input types and arguments which I would argue that would not be necessary for now.

Reply:
> Unless you want to have different rules,

Yeah, that was exactly the thought. For now I've been trying to make the rules do as little as possible such that people can customize which ones they want to use.

After writing this I noticed we already have `input-object-values-have-descriptions` which validates both `input { .. }` and arguments so I guess this one is consistent with the behavior. 

My bad.

Reply:
Yes, I followed the styles of the descriptions rules in spirit. I think I could do another PR breaking the input-object-values check in two rules. That would make more sense to me.

Reply:
I think we can keep this as-is for now.

If someone needs more granular rules for `input-object-values`, we can split the them into different rules for arguments vs input fields. If we do this, we should do it in such a way that we don't change the behavior and break someone's existing workflow. One idea is to keep the old rules around, but deprecate them and remove them from the documentation.

But, again, I think what you did is fine and consistent with the current state. I'll 🚢 this shortly.